velox/exec/benchmarks/HashJoinListResultBenchmark.cpp

```diff
@@ -0,0 +1,465 @@
+        mode == BaseHashTable::HashMode::kArray) {
+      VELOX_FAIL("Bad hash mode.");
+    }
+
+    numDependentFields = buildType->size() - 1;
+    if (mode == BaseHashTable::HashMode::kNormalizedKey) {
+      extraValue = (2L << 20) + 100;
+    } else if (mode == BaseHashTable::HashMode::kHash) {
+      extraValue = std::numeric_limits<int64_t>::max() - 1;
+    } else {
+      extraValue = 0;
+    }
+    std::string modeString = mode == BaseHashTable::HashMode::kArray ? "array"
+        : mode == BaseHashTable::HashMode::kHash                     ? "hash"
+                                                 : "normalized key";
+    std::stringstream distStr;
+    for (auto dist : keyRepeatTimesDistribution) {
+      distStr << fmt::format("{}%:{};", dist.first, dist.second);
+    }
+    title = fmt::format(
+        "{}_{}_{}", modeString, numDependentFields + 1, distStr.str());
+  }
+
+  // Title for reporting
+  std::string title;
+
+  // Expected mode.
+  BaseHashTable::HashMode mode;
+
+  // Type of build & probe row.
+  TypePtr buildType{ROW({"k1"}, {BIGINT()})};
+
+  // Distinct rows in the table.
+  int64_t hashTableSize;
+
+  // Number of probe rows.
+  int64_t probeSize;
+
+  // Number of build rows.
+  int64_t buildSize;
+
+  // This is used to control the hash mode.
+  int64_t extraValue;
+
+  std::vector<std::pair<int32_t, int32_t>> keyRepeatTimesDistribution;
+
+  std::vector<std::pair<int32_t, int32_t>> buildKeyRepeat;
+
+  int32_t numDependentFields;
+
+  int32_t numTables{7};
+
+  std::string toString() const {
+    std::string modeString = mode == BaseHashTable::HashMode::kArray ? "array"
+        : mode == BaseHashTable::HashMode::kHash                     ? "hash"
+                                                 : "normalized key";
+    std::stringstream distStr;
+    for (auto dist : keyRepeatTimesDistribution) {
+      distStr << fmt::format("{}%:{}; ", dist.first, dist.second);
+    }
+    return fmt::format(
+        "HashTableSize:{}, BuildInputSize:{}, ProbeInputSize:{}, ExpectHashMode:{}, KeyRepeatTimesDistribution: {}",
+        hashTableSize,
+        buildSize,
+        probeSize,
+        modeString,
+        distStr.str());
+  }
+};
+
+struct HashTableBenchmarkRun {
+  HashTableBenchmarkParams params;
+
+  int64_t numOutput;
+
+  int32_t numIter{1};
+
+  // The mode of the table.
+  BaseHashTable::HashMode hashMode;
+
+  void merge(HashTableBenchmarkRun other) {
+    numIter++;
+  }
+
+  std::string toString() const {
+    std::stringstream out;
+    out << params.toString();
+
+    std::string modeString = hashMode == BaseHashTable::HashMode::kArray
+        ? "array"
+        : hashMode == BaseHashTable::HashMode::kHash ? "hash"
+                                                     : "normalized key";
+    out << std::endl << " mode=" << modeString << " numOutput=" << numOutput;
+    return out.str();
+  }
+};
+
+class HashTableListJoinResultBenchmark : public VectorTestBase {
```

Comment:
> public VectorTestBase

Do you need this inheritance? 

Looks like you are using VectorMaker directly.

```
vectorMaker_->flatVector<int64_t>(data)
```

Since you use VectorTestBase as a base class, the above would be simpler as 

```
makeFlatVector<int64_t>(data)
```